lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1251,18 +1251,25 @@ def _process_logging_relation_changed(self, relation: Relation):
 
         Args:
             relation: the `Relation` instance to update.
+            event: an optional event, the type of which is checked. If it is a
+                RelationDepartedEvent no updates are sent to the relation data bag
+
         """
         if self._charm.unit.is_leader():
+            # We don't need to set a bunch of data if a relation is going away
             relation.data[self._charm.app].update(self._promtail_binary_url)
             logger.debug("Saved promtail binary url: %s", self._promtail_binary_url)
```

Comment:
`_promtail_binary_url` should probably go into relation created (see  #134).

Reply:
Completely unrelated to this patch.